bazel/external/quiche.BUILD

```diff
@@ -2399,7 +2386,6 @@ envoy_cc_library(
     tags = ["nofips"],
     visibility = ["//visibility:public"],
     deps = [
-        ":quic_core_buffer_allocator_lib",
```

Comment:
Since some of these source files (like quic_frame.cc) include quiche_buffer_allocator.h, should this target not depend directly on :quiche_common_buffer_allocator_lib?  (I know that it does transitively via :quic_platform_base.)  Same question for :quic_core_http_capsule_lib and :quic_test_tools_test_utils_lib, and possibly targets in other BUILD files where :quic_core_buffer_allocator_lib dependency is removed.

Reply:
Oh, good point. I'm not sure how they got removed. Maybe that was part of the PR I based this on. Anyway, I added them back. Thanks!